services/QuillLMS/app/workers/copy_concept_results_to_responses_worker.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class CopyConceptResultsToResponsesWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: SidekiqQueue::MIGRATION
+
+  BATCH_SIZE=100000
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  def perform(start, finish)
+    directions_cache = {}
+    instructions_cache = {}
+    previous_feedbacks_cache = {}
+    prompts_cache = {}
+    question_types_cache = {}
+
+    Response.bulk_insert(ignore: true) do |worker|
+      ConceptResult.where(id: start..finish).each do |concept_result|
+
+        directions = concept_result.metadata['directions']
+        instructions = concept_result.metadata['instructions']
+        previous_feedback = concept_result.metadata['lastFeedback']
+        prompt = concept_result.metadata['prompt']
+        question_type = concept_result.question_type
+
+        directions_cache[directions] ||= ResponseDirections.find_or_create_by(text: directions)&.id if directions
```

Comment:
Are we concerned with any `find_or_create_by` silently failing?  I realize the data is already in the database, but the unknown unknowns always lurk, especially with so many records..  Maybe `add a find_or create_by!`  and a corresponding `rescue` clause that documents the failed creation?

Reply:
What would a failure look like?  Are you imagining that it somehow fails to find, tries to do a create, but then the `create` fails?  That seems like a reasonable thing to worry about.  I think we could just get away with the just making it `find_or_create_by!` and not rescuing.  As long as we're monitoring the retry queue, we should be able to keep things cleaned up.

Reply:
That sounds good.  I just realized that `rescue` clauses add a performance penalty, and given the size of this migration, it's probably better to just monitor the retry queue like you suggest.